client/src/components/Stalker.svelte

```diff
@@ -0,0 +1,106 @@
+<script lang="ts">
+  import type { ChatRoom as ChatRoomType, ChatMessage, Lobby } from "$lib/firebase/firestore-types/lobby";
+  import { stalkChatroom } from "$lib/firebase/firestore-functions";
+  import { getChatRoomCollection, getChatRoomMessagesCollection } from "$lib/firebase/firestore-collections";
+  import { onSnapshot, query, orderBy, collection } from "firebase/firestore";
+  import { onMount, onDestroy } from "svelte";
+  import type { Unsubscribe } from "firebase/auth";
+  // import { auth } from "$lib/firebase/app";
+  import { authStore as user } from "$stores/auth";
+  import ChatRoom from "./ChatRoom.svelte";
+
+  export let lobby: Lobby;
+  export let lobbyCode: string;
+
+  let unsubscribeChatrooms: Unsubscribe | undefined = undefined;
+  let chatrooms: ChatRoomType[] = [];
+  let chatroomsIds: string[] = [];
+  let chatMessages: ChatMessage[] = [];
+
+  onMount(async () => {
+    const chatCollection = getChatRoomCollection(lobbyCode);
+    unsubscribeChatrooms = onSnapshot(chatCollection, (chatSnapshot) => {
```

Comment:
do we need to be subscribed to the chatrooms collection? Will there be any changes we need to listen for?